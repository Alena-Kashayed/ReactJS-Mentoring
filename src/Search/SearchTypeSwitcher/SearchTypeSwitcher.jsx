import React from 'react';
import styles from './SearchTypeSwitcher.scss';
import classNames from 'classnames/bind';

function SearchTypeSwitcher(props) {
    const cx = classNames.bind(styles);

    return (
        <div className={styles.SearchTypeSwitcher}>
            <div className={styles.searchByGroup}>
                <span className={styles.searchBy}>Search by</span>
                <button className={cx('searchByTitle', 'searchByBtn')}>
                    Title
                </button>
                <button className={cx('searchByDirector', 'searchByBtn', 'active')}>
                    Director
                </button>
            </div>
            <button className={cx('search', 'searchByBtn')}>
                Search
            </button>
        </div>
    );
}
export default SearchTypeSwitcher;
