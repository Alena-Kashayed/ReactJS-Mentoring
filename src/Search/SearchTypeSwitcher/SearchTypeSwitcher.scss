.SearchTypeSwitcher {

  .searchByGroup {
    min-width: 250px;
    display: flex;
    justify-content: space-between;
    align-items: baseline;

    .searchBy {
      text-transform: uppercase;
      color: #FFFFFF;
      font-size: 12px;
    }

    .active {
      background-color: #3C3C3C;
      cursor: auto;
    }
  }

  .searchByBtn {
    background-color: #F55263;
    border: none;
    border-radius: 2px;
    color: #FFFFFF;
    padding: 3px 12px;
    cursor: pointer;
    text-transform: uppercase;
    font-size: 12px;
  }
}

