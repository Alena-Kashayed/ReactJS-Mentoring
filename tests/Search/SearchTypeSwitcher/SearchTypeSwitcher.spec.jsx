import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchTypeSwitcher from '../../../src/Search/SearchTypeSwitcher/SearchTypeSwitcher';

describe('SearchTypeSwitcher', () => {
  it('is rendered correctly', () => {
    const typeOfQuery = 'movie';
    const wrapper = mount(<SearchTypeSwitcher
      typeOfQuery={typeOfQuery}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
