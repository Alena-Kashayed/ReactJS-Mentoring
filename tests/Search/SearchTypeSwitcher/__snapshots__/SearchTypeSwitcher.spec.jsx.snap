// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTypeSwitcher is rendered correctly 1`] = `
<div
  className="SearchTypeSwitcher"
>
  <div
    className="searchByGroup"
  >
    <span
      className="searchBy"
    >
      Search by
    </span>
    <button
      className="searchByBtn active"
      onClick={[Function]}
    >
      Movie
    </button>
    <button
      className="searchByBtn"
      onClick={[Function]}
    >
      TV show
    </button>
  </div>
</div>
`;
